src/components/TemplateList/TemplateList.js

```diff
@@ -259,13 +259,23 @@ const TemplateList = ({
           let file = url;
           let headers = new Headers();
           headers.append('Authorization', bearerToken);
-          const fileName = `${studyKfid ? studyKfid : 'data'}_templates${
+          var fileName = `${studyKfid ? studyKfid : 'data'}_templates${
             format === 'zip' ? '.zip' : '.xlsx'
           }`;
           fetch(file, {headers})
-            .then(response => response.blob())
+            .then(response => {
+              fileName = response.headers
+                .get('Content-Disposition')
+                .split('filename*=UTF-8')[1]
+                .substring(2);
+              response.blob();
+            })
             .then(blobby => {
-              let objectUrl = window.URL.createObjectURL(blobby);
+              var binaryData = [];
+              binaryData.push(blobby);
+              let objectUrl = window.URL.createObjectURL(
+                new Blob(binaryData, {type: 'application/zip'}),
```

Comment:
Will this always create a zip?

Reply:
Tested download of an excel workbook, and it doesn't open bc it thinks file is corrupted. Probably bc its downloaded as a zip